I found that when I merely adjusted the dependency on embedded-hal to 1.0.0 it led to build failures in src/compat.rs:

```
error[E0277]: the trait bound `<T as embedded_hal::digital::v2::OutputPin>::Error: embedded_hal::digital::Error` is not satisfied
  --> src/compat.rs:22:18
   |
22 |     type Error = T::Error;
   |                  ^^^^^^^^ the trait `embedded_hal::digital::Error` is not implemented for `<T as embedded_hal::digital::v2::OutputPin>::Error`
   |
note: required by a bound in `embedded_hal::digital::ErrorType::Error`
  --> /Users/stelterd/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/digital.rs:61:17
   |
61 |     type Error: Error;
   |                 ^^^^^ required by this bound in `ErrorType::Error`
help: consider further restricting the associated type
   |
20 |     T::Error: fmt::Debug, <T as embedded_hal::digital::v2::OutputPin>::Error: embedded_hal::digital::Error
   |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

It wasn't obvious to me what to do about that, which is why I just started deleting stuff. 

As for whether it's relevant to still support embedded-hal 0.2.x, the maintainers seem to think so, there's an embedded-hal-compat crate to ease interoperability: https://github.com/rust-embedded/embedded-hal/blob/master/docs/migrating-from-0.2-to-1.0.md#embedded-hal-compat

If you can see how to work around the above issues, I certainly don't object to keeping the compat code in place! Maybe it would make sense to guard it with a Cargo config macro, so it conditionally compiles only when needed? I'm not sure if a Crate feature is the appropriate way to handle that or some other mechanism. I can take a stab at it if that seems like the right course!

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,7 +30,7 @@ categories = ["embedded", "hardware-support", "no-std", "science::robotics"]
 
 
 [dependencies]
-embedded-hal = "=1.0.0-alpha.9"
+embedded-hal = "1.0.0"
 fugit = "0.3.5"
 fugit-timer = "0.1.3"
 nb = "1.0.0"

--- a/src/compat.rs
+++ b/src/compat.rs
@@ -43,6 +43,6 @@ where
     }
 
     fn set_high(&mut self) -> Result<(), Self::Error> {
-        self.0.set_low().map_err(CompatError)
+        self.0.set_high().map_err(CompatError)
     }
 }

--- a/src/compat.rs
+++ b/src/compat.rs
@@ -1,9 +1,8 @@
 //! Compatibility code to help use Stepper on more platforms
 
 use core::fmt;
-
-use embedded_hal::digital::ErrorType;
 use embedded_hal::digital::OutputPin;
+use embedded_hal::digital::{Error, ErrorKind, ErrorType};
 use embedded_hal_stable::digital::v2::OutputPin as StableOutputPin;
 
 /// Wrapper around a pin
@@ -14,12 +13,24 @@ use embedded_hal_stable::digital::v2::OutputPin as StableOutputPin;
 /// `embedded-hal`.
 pub struct Pin<T>(pub T);
 
+/// Wrapper for error compatibility
+#[derive(Debug)]
+pub struct CompatError<T>(pub T);
+
+impl<T> Error for CompatError<T>
+    where T: fmt::Debug
+{
+    fn kind(&self) -> ErrorKind {
+        ErrorKind::Other
+    }
+}
+
 impl<T> ErrorType for Pin<T>
 where
     T: StableOutputPin,
     T::Error: fmt::Debug,
 {
-    type Error = T::Error;
+    type Error = CompatError<T::Error>;
 }
 
 impl<T> OutputPin for Pin<T>
@@ -28,10 +39,10 @@ where
     T::Error: fmt::Debug,
 {
     fn set_low(&mut self) -> Result<(), Self::Error> {
-        self.0.set_low()
+        self.0.set_low().map_err(CompatError)
     }
 
     fn set_high(&mut self) -> Result<(), Self::Error> {
-        self.0.set_high()
+        self.0.set_low().map_err(CompatError)
     }
 }

